p2p/peer/peerstore.go

```diff
@@ -73,35 +93,57 @@ func (ab *addressbook) Addresses(p ID) []ma.Multiaddr {
 		return nil
 	}
 
-	maddrs2 := make([]ma.Multiaddr, 0, len(maddrs))
-	for _, m := range maddrs {
-		maddrs2 = append(maddrs2, m)
+	good := make([]ma.Multiaddr, 0, len(maddrs))
+	var expired []string
+	for s, m := range maddrs {
+		if m.Expired() {
+			expired = append(expired, s)
+		} else {
+			good = append(good, m.Addr)
+		}
+	}
+
+	// clean up the expired ones.
+	for _, s := range expired {
+		delete(ab.addrs[p], s)
```

Comment:
yep, it is. defered. github's omitting it from the diff. (but would've been a great catch)